ConfigSpace/util.py

```diff
@@ -81,137 +81,157 @@ def get_one_exchange_neighbourhood(configuration: Configuration, seed: int) -> L
     In: Proceedings of the conference on Learning and Intelligent OptimizatioN (LION 5)
     """
     random = np.random.RandomState(seed)
-    neighbourhood = []
-    for i, hp_name in enumerate(configuration):
-        number_of_sampled_neighbors = 0
-        array = configuration.get_array()
-
-        if not np.isfinite(array[i]):
-            continue
+    hyperparameters_list = list(configuration.keys())
+    hyperparameters_list_length = len(hyperparameters_list)
+    neighbors_to_return = dict()
+
+    # neighbourhood = []
+    # for i, hp_name in enumerate(configuration):
+    for i in range(hyperparameters_list_length):
+        index = np.random.randint(hyperparameters_list_length)
+        hp_name = hyperparameters_list[index]
+        if hp_name in neighbors_to_return:
+            np.random.shuffle(neighbors_to_return[hp_name])
+            n_ = neighbors_to_return[hp_name].pop()
+            if len(neighbors_to_return[hp_name]) == 0:
+                del neighbors_to_return[hp_name]
```

Comment:
Don't delete here. If the list is empty, then you need to sample a different hyperparameter.